File: c++/2-conditionals-and-loops/for/lesson.cpp
// normal for loop
#include <iostream>
using namespace std;

int main() {
    for(int i=1;i<=5;i++) {
        cout << i << endl;
    }
}

// output: 1, 2, 3, 4, 5




// for loop with `break;` --> exits the loop when condition is met
#include <iostream>
using namespace std;

int main() {
    for(int i=0;i<10;i++) {
        if(i==5) {
            break;
        }
        cout << i << endl;
    }
}

// output: 0,1,2,3,4



// for loop with `continue;` -->  The continue statement skips the current loop iteration and continues with the next one.
#include <iostream>
using namespace std;

int main() {
    for(int i=0;i<10;i++) {
        if(i==5) {
            continue;
        }
        cout << i << endl;
    }
}

// output: 0,1,2,3,4,6,7,8,9


//combination
#include <iostream>
using namespace std;

int main() {
    for(int x=1;x<10;x++) {
      if(x == 2) {
        continue; 
      }
      if(x == 5) {
        break;
      }
      cout << x << endl;
    }
}

// output: 1, 3, 4

File: python/4-list/3-for/lesson.py
nums = [1, 2, 3, 4]

res = 0

for x in nums:

    if x % 2 == 0:

        continue
    else:
        res += x

print(res)


"""
This code defines a list of numbers called "nums" and a variable "res" that is initially set to 0. It then iterates through each number in the list "nums" using a for loop. For each iteration, it checks if the current number (x) is even (i.e. x%2==0) using an if statement. If the number is even, the loop continues to the next iteration using the "continue" statement. If the number is odd, the "res" variable is incremented by the value of x using the "res += x" statement. Finally, the code prints the final value of "res" which will be the sum of all odd numbers in the list "nums".

The output for the above code would be 4 because the list nums contains [1,2,3,4], out of these numbers 1 and 3 are odd numbers, and the sum of 1 and 3 is 4.

"""
